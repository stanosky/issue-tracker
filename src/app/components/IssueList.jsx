import React from "react";
import { IssueListItem } from "./IssueListItem";

export const IssueList = ({ issues, name }) => (
  <div className="col-md-4 card p-2 mt-2">
    <h2>{name}</h2>
    <div>
      {issues.map((issue) => (
        <IssueListItem {...issue} key={issue.id} />
      ))}
    </div>
  </div>
);
