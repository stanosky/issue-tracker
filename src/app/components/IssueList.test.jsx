import React from "react";
import { shallow } from "enzyme";
import { IssueList } from "./IssueList";

describe("IssueList", () => {
  it("should render without errors", () => {
      const props = {
          issues: [
              {
                  id: 'id 1',
                  name: 'name 1',
                  desc: 'desc 1',
                  group: 'group 1',
              },
              {
                  id: 'id 2',
                  name: 'name 2',
                  desc: 'desc 2',
                  group: 'group 1',
              }
          ],
          name: 'group 1'
      }
    const wrapper = shallow(<IssueList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
