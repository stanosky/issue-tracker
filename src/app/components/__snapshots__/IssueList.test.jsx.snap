// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueList should render without errors 1`] = `
<div
  className="col-md-4 card p-2 mt-2"
>
  <h2>
    group 1
  </h2>
  <div>
    <IssueListItem
      desc="desc 1"
      group="group 1"
      id="id 1"
      key="id 1"
      name="name 1"
    />
    <IssueListItem
      desc="desc 2"
      group="group 1"
      id="id 2"
      key="id 2"
      name="name 2"
    />
  </div>
</div>
`;
